pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -95,8 +95,13 @@ public void init(File dataFile, Set<String> fieldsToRead, @Nullable RecordReader
     _recordExtractor = new CSVRecordExtractor();
     CSVRecordExtractorConfig recordExtractorConfig = new CSVRecordExtractorConfig();
     recordExtractorConfig.setMultiValueDelimiter(multiValueDelimiter);
-    _recordExtractor.init(fieldsToRead, recordExtractorConfig);
+
     init();
+
+    if (fieldsToRead == null || fieldsToRead.isEmpty()) {
```

Comment:
i see that in CSV case where fieldsToRead is null, we get fields to read in the RecordReader. Versus, in the JSON/AVRO case we used to make that decision inside the RecordExtractor.
Is it possible to keep these consistent and always let the RecordExtractor make this decision?

Reply:
I had the same thought and was debating whether or not to follow the same pattern as the other extractors. I eventually decided to put the "read all fields" in the CSVRecordReader because the field names are accessible only through the CSV header and not in the record object being passed to the `extract` method.

Reply:
I prefer the alternate implementation. Reason being, the RecordExtractor is a public API. They could be used in external applications. The external users should not have to know that all except CSVRecordExtractor expect fieldsToRead upfront.

Reply:
After syncing over Slack, we decided to keep the decision to extract all fields within the extractor in case any other client directly calls the extractor.